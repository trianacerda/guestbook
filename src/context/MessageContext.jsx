import { useContext, createContext, useState } from 'react';

const MessageContext = createContext();

const MessageProvider = ({ children }) => {
  const [messages, setMessages] = useState('');

  return (
    <MessageContext.Provider value={{ messages, setMessages }}>
      {children}
    </MessageContext.Provider>
  );
};

const useMessages = () => {
  const context = useContext(MessageContext);

  if (context === undefined) {
    throw new Error(
      'useMessages hook must be used withing a MessageContext Provider parent tag'
    );
  }

  return context;
};

export { MessageProvider, useMessages };
