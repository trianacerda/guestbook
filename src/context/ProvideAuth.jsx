import { createContext, useState } from 'react';

export const authContext = createContext();

export function ProvideAuth({ children }) {
  const [user, setUser] = useState(null);

  const login = (email, password) => {
    const loginSuccess =
      email === process.env.AUTH_EMAIL &&
      password === process.env.AUTH_PASSWORD;
    if (loginSuccess) setUser({ email });
    return loginSuccess;
  };

  const logout = (callback) => {
    setUser(null);
    callback();
  };

  return (
    <authContext.Provider value={{ user, login, logout }}>
      {children}
    </authContext.Provider>
  );
}
