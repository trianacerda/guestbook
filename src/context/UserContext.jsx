import React from 'react';
import { useContext, createContext, useState } from 'react';

const UserContext = createContext();

const UserProvider = ({ children }) => {
  const [user, setUser] = useState('');

  return (
    <UserContext.Provider value={{ user, setUser }}>
      {children}
    </UserContext.Provider>
  );
};

const useUser = () => {
  const context = useContext(UserContext);

  if (context === undefined) {
    throw new Error(
      'useUser hook must be used withing a MessageContext Provider parent tag'
    );
  }

  return context;
};

export { UserProvider, useUser };
