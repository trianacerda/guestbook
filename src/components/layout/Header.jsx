import React from 'react';
import { useUser } from '../../context/UserContext';

const Header = () => {
  const { user } = useUser();

  return (
    <header>
      {user ? (
        <p>
          You're Signing Guestbook as
          <span className="header-user">{user}</span>
        </p>
      ) : (
        <p>Aw Shucks-- Please Sign In!</p>
      )}
    </header>
  );
};

export default Header;
