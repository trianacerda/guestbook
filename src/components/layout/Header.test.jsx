import React from 'react';
import { render, screen } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import Header from './Header';
import { UserProvider } from '../../context/UserContext';

it('should render a LayoutViews + children components', async () => {
  const { container } = render(
    <MemoryRouter>
      <UserProvider>
        <Header />
      </UserProvider>
    </MemoryRouter>
  );

  const header = await screen.findByText('Aw Shucks-- Please Sign In!');

  expect(header).toBeInTheDocument();
  expect(container).toMatchSnapshot();
});
