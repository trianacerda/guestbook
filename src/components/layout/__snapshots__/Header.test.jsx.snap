// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render a LayoutViews + children components 1`] = `
<div>
  <header>
    <p>
      Aw Shucks-- Please Sign In!
    </p>
  </header>
</div>
`;
